### PR TITLE
Fix run status UI and progress presentation

### PR DESCRIPTION
**Pull request by gtpooniwala**
## Summary
- move the primary run status UI out of the top context strip into a dedicated status row above the composer so it stays stable in both empty and full conversations
- derive user-facing run progress from transport state and recent run events instead of showing raw status strings like `Run running`
- add focused frontend coverage for the new run-status presentation and degraded transport behavior

## Linked Issue
Refs #187

## Validation
- [x] `cubic review --base main`
- [x] `npm test -- --runTestsByPath tests/ChatPanel.test.jsx tests/runStatus.test.js tests/sendMessage.test.jsx tests/formatters.test.js`
- [ ] `python -m unittest discover -s tests -p "test_*.py" -v`
- [ ] `python tests/run_repo_checks.py`
- [ ] Additional evals required for LLM/tool-calling behavior changes

## Workflow Checklist
- [x] Branch created in isolated worktree (not `main`)
- [x] Rebasing done against latest `origin/main` before push/PR
- [x] Commits are granular and focused
- [ ] CI checks pass
- [x] Merge method will be **Squash and merge**

---
## Summary by cubic
Moved run status to a dedicated bar above the composer with clear, human labels based on transport and recent events. Adds transport-aware states (streaming, polling, degraded) with helpful messages and unifies labels/tones across chat, dashboard, and list. Refs #187.

- **New Features**
  - Dedicated run status row with a pill and detail text; ARIA live; kept separate from transcript and document chips.
  - Transport-aware progress: streaming/polling/degraded with messages like “Updates lost” and “Live stream unavailable. Checking status in the background.” Short labels in the conversation list; dashboard shows label + detail.
  - New `.degraded` and pill styles; flex layout and nowrap tweaks to prevent overflow and keep panels flexible.

- **Refactors**
  - Centralized run presentation in `@/lib/runStatus` (`getRunPresentation`, `getRunStatusClassName`, `RUN_IN_PROGRESS_STATUSES`) for consistent tone mapping.
  - Workspace tracks `transport` and `transportMessage`; SSE fallback switches to `polling`; timeouts mark runs `degraded`; completion marks `complete`; polling tolerates missing `status` fields.
  - Expanded tests for run-status presentation/layout, degraded/polling messages, and SSE fallback edge cases.

<sup>Written for commit 91cfb69fe3a0acfd66000e34fa656b79676e378e. Summary will update on new commits.</sup>

### CODE REVIEW

File: frontend/lib/runStatus.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    case "cancelled":
+      label = "Cancelled";
+      shortLabel = "Cancelled";
+      detail = "The latest run was cancelled.";
+      break;
+    default:
+      break;
+  }
+
+  if (latestEvent?.type === "queued") {
```
[the remark this answers is not in the document]
Comment (gtpooniwala) [reply frontend/lib/runStatus.js]: cubic found a real issue here. I now guard the `queued` / `started` / `retrying` latest-event label overrides behind `!isTerminal`, so terminal runs keep their terminal labels. I also added a regression test for a succeeded run whose latest event is still `started`.

File: frontend/components/WorkspaceApp.jsx [hunk 2]
```diff
@@ -540,6 +541,10 @@ export default function WorkspaceApp({ view, currentPath, initialConversationId
                 runId: pollRunId,
                 status: pollStatus?.status || previousRunState.status || "queued",
                 error: pollStatus?.error || "",
+                transport: RUN_IN_PROGRESS_STATUSES.has(pollStatus?.status) ? "polling" : "complete",
+                transportMessage: RUN_IN_PROGRESS_STATUSES.has(pollStatus?.status)
+                  ? "Live stream unavailable. Checking status in the background."
+                  : "",
```
[the remark this answers is not in the document]
Comment (gtpooniwala) [reply frontend/components/WorkspaceApp.jsx]: Fixed. `pollRunUntilComplete()` now resolves one effective status value per poll cycle and uses that same resolved value for `status`, transport mode, transport message, and the break condition. I also added a regression test covering a polling fallback where the status payload temporarily omits `status`.
